guides/03_building-with-blocks/06_theming-guide

```diff
@@ -0,0 +1,177 @@
+# Theming
+Tags: THEMES
+
+## Introduction
+
+Gradio features a built-in theming engine that lets you customize the look and feel of your app. You can choose from a variety of themes, or create your own. To do so, pass the `theme=` kwarg to the `Blocks` or `Interface` constructor. For example:
+
+```python
+with gr.Blocks(theme=gr.themes.Glass()):
+    ...
+```
+
+Gradio comes with a set of prebuilt themes which you can load from `gr.themes.*`. Each of these themes set values for hundreds of CSS variables.
+
+You can use prebuilt themes as a starting point for your own custom themes, or you can create your own themes from scratch. Let's take a look at each approach.
+
+## Extending Themes via the Constructor
+
+Although each theme has hundreds of CSS variables, the values for most these variables are drawn from 8 core variables which can be set through the constructor of each prebuilt theme. Modifying these 8 arguments allows you to quickly change the look and feel of your app.
+
+### Core Colors
+
+The first 3 constructor arguments set the colors of the theme and are gradio.themes.Color objects. Internally, these Color objects hold brightness values for the palette of a single hue, ranging from 50, 100, 200..., 800, 900, 950. Other CSS variables are derived from these 3 colors.
+
+The 3 color constructor arguments are:
+- `primary_hue`: This is the color draws attention in your theme. In the default theme, this is set to gradio.themes.Orange.
+- `secondary_hue`: This is the color that is used for secondary elements in your theme. In the default theme, this is set to gradio.themes.Blue.
```

Comment:
```suggestion
- `secondary_hue`: This is the color that is used for secondary elements in your theme. In the default theme, this is set to `gradio.themes.Blue`. It can also be specified as a string color.
```